packages/apollo-datasource-rest/src/RESTDataSource.ts

```diff
@@ -282,7 +282,7 @@ export abstract class RESTDataSource<TContext = any> extends DataSource {
     request: Request,
     fn: () => Promise<TResult>,
   ): Promise<TResult> {
-    if (process && process.env && process.env.NODE_ENV === 'development') {
+    if (process?.env?.NODE_ENV === 'development') {
```

Comment:
@benjamn Wow, `__DEV__` looks super neat 👍 
I would love to use it in `graphql-js` since it will make `graphql-js` Deno compatible.
I still can do this change since `16.0.0` is not released yet.
Do you see any potential issues with switching `graphql-js` to `__DEV__`?

Reply:
Yes, I think `__DEV__` could work for `graphql-js` too, especially since we're only talking about this one spot:
```sh
~/dev/graphql-js % grep -r process.env src 
src/jsutils/instanceOf.ts:  process.env.NODE_ENV === 'production'
```
I've learned some things from working with these globals recently, so I'd be happy to submit a `graphql-js` PR that takes both `process.env.NODE_ENV` and `__DEV__` into account, in a safe way (without polyfilling).

Reply:
@benjamn Sounds great 👍
Would love to have this PR.

Reply:
Thanks for sharing this @benjamn !

Reply:
Note that we do use `process.env` for a lot more than `NODE_ENV` in Apollo Server, though the exact mechanism of reading the env vars could be extracted out.